hedera-node/hapi-fees/src/main/java/com/hedera/node/app/hapi/fees/usage/crypto/CryptoOpsUsage.java

```diff
@@ -122,12 +122,23 @@ public long cryptoAutoRenewRb(final ExtantCryptoContext ctx) {
                 + ctx.currentNumTokenRels() * CRYPTO_ENTITY_SIZES.bytesInTokenAssocRepr();
     }
 
+    /**
+     * Returns the estimated resource usage for a crypto update transaction.
+     *
+     * @param sigUsage the already-computed resource usage for signature verification
+     * @param baseMeta the already-computed resource usage for the base transaction
+     * @param cryptoUpdateMeta metadata summarizing the update transaction
+     * @param ctx the current state of the crypto account
+     * @param accumulator the resource usage accumulator
+     * @param explicitAutoAssocSlotLifetime if non-zero, the lifetime to use for resource usage of new auto-renew slots
```

Comment:
actually only if positive.  if negative doesn't work this way (due to `Math.max`).  maybe on entry check the precondition ≥0?  (Or just hi pass at 0.)

Reply:
Updated javadoc to be more precise.